pkg/sharky/shard.go

```diff
@@ -5,18 +5,79 @@
 package sharky
 
 import (
+	"encoding/binary"
 	"encoding/json"
-	"os"
+	"errors"
+	"io"
 	"sync"
 )
 
+// size of the byte representation of Location
+const LocationSize int = 17
+
 // location models the location <shard, offset, length> of a chunk
 type Location struct {
 	Shard  uint8
 	Offset int64
 	Length int64
 }
 
+// MarshalBinary returns byte representation of location. We use binary.PutVarint
+// which can take upto 10bytes to marshal int64. Realistically, we will never have
+// values here that would require extra bytes to pack. The offset is currently capped
+// and the length is deterministic. In order to save the extra bytes, we use the min
+// required bytes here which is 8
```

Comment:
To be completely correct that capping extra two bytes does not make provide any possibility for error, I would suggest to use uint64 for Location.Offset and Location.Length and to encode them with PutUint64 which already requires 8 bytes and both offset and length values should not be negative.